content/docs/projects/tdd.md

```diff
@@ -0,0 +1,602 @@
+        for customer in customers:
+            TestCustomer.destroy(customer)
+        for uom in uoms:
+            TestUOM.destroy(uom)
+```
+
+## 5.2 Unit Tests
+
+The unit testing framework that we use at BizKit is unittest. Read its official documentation [here](https://docs.python.org/3/library/unittest.html).
+
+### 5.2.1 Red, Green, Refactor
+
+#### 5.2.1.1 Getting Started
+
+The red, green, refactor approach is a common technique used in test-driven development. It consists of the following steps:
+
+1. <span style="color: red;">Red</span>: Write a failing test for a piece of functionality that you want to add to your code. The test should define the expected behavior of the code and should be written before the code itself is written.
+2. <span style="color: green;">Green</span>: Write the code to make the test pass. This should be done in a way that satisfies the requirements of the test and ensures that the code works as expected. This is typically the minimum amount of code needed to make the test pass.
+3. <span style="color: blue;">Refactor</span>: Once the code passes the test, you can refactor the code to improve its design, readability, and performance. This is done without changing the functionality of the code, so the tests should still pass.
+
+The red-green-refactor approach is useful because it helps you focus on one small piece of functionality at a time, which makes the development process more manageable. It also encourages you to think about the design of your code upfront, and helps ensure that your code is well-tested.
+
+#### 5.2.1.2 Example
+
+Here is an example of the red, green, refactor approach in test-driven development:
+
+1. <span style="color: red;">Red</span>: Write a test to check that a function called `add` can be used to add two numbers together and return the result. The test should define the expected behavior of the `add` function, and should be written before the `add` function itself is written.
+
+```python
+def test_add():
+  result = add(2, 3)
+  assert result == 5
+```
+
+2. <span style="color: green;">Green</span>: Write the `add` function to make the test pass. This should be done in a way that satisfies the requirements of the test and ensures that the `add` function works as expected.
+
+```python
+def add(a, b):
+  return 5
+```
+
+3. <span style="color: blue;">Refactor</span>: Once the `add` function passes the test, you can refactor the code to make it more modular and readable. This might involve restructuring the code, adding comments, or making other improvements.
+
+```python
+def add(a, b):
+  # Return the sum of a and b
+  return a + b
+```
+
+In this example, the red, green, refactor approach is used to write a simple function called `add` that can be used to add two numbers together and return the result. The test is written first, and then the code is written to make the test pass. Finally, the code is refactored to improve its readability and maintainability.
```

Comment:
Might also be good to note here that this is an iterative approach, where after refactoring, you would normally add more tests or update your tests to handle more scenarios. 

In the provided example, you might want to add more scenarios to the add test, like maybe looping through a list of different number cominations to add.

Reply:
856cf12959eea44b2307f78bbf6ce9b83a24a51b